packages/insured-bridge-relayer/test/Relayer.ts

```diff
@@ -1344,4 +1357,78 @@ describe("Relayer.ts", function () {
+      await bridgePool.methods.addLiquidity(initialPoolLiquidity).send({ from: l1LiquidityProvider });
+
+      // Mint some tokens for the L2 depositor and approve. Mint 2x for the two deposits to be batched.
+      await l2Token.methods.mint(l2Depositor, toBN(depositAmount).muln(2)).send({ from: l2Owner });
+      await l2Token.methods
+        .approve(bridgeDepositBox.options.address, toBN(depositAmount).muln(2))
+        .send({ from: l2Depositor });
+    });
+    it("Can correctly send batch relays and settle relay with multicall", async function () {
+      // Before any relays should do nothing and log accordingly.
+      await Promise.all([l1Client.update(), l2Client.update()]);
+      await relayer.checkForPendingDepositsAndRelay();
+      assert.isTrue(lastSpyLogIncludes(spy, "No relayable deposits"));
+
+      // Make two deposits on L2 and check the bot relays them in a batch accordingly.
+      const relayTime = await bridgePool.methods.getCurrentTime().call();
+      const quoteTime = (await web3.eth.getBlock("latest")).timestamp;
+      // Deposit #1
+      await bridgeDepositBox.methods
+        .deposit(
+          l2Depositor,
+          l2Token.options.address,
+          depositAmount,
+          defaultSlowRelayFeePct,
+          "0", // set to zero to force slow relay for this test.
+          quoteTime
+        )
+        .send({ from: l2Depositor });
+
+      // Deposit #2
+      await bridgeDepositBox.methods
+        .deposit(
+          l2Depositor,
+          l2Token.options.address,
+          depositAmount,
+          defaultSlowRelayFeePct,
+          "0", // set to zero to force slow relay for this test.
+          quoteTime
+        )
+        .send({ from: l2Depositor });
+
+      await Promise.all([l1Client.update(), l2Client.update()]);
+
+      await l1Token.methods.mint(l1Relayer, toBN(depositAmount).muln(4)).send({ from: l1Owner });
+      await l1Token.methods.approve(bridgePool.options.address, toBN(depositAmount).muln(4)).send({ from: l1Relayer });
+      await relayer.checkForPendingDepositsAndRelay();
+
+      // The log should also inform that this is a multicall batch.
+      assert.isTrue(lastSpyLogIncludes(spy, "Multicall Transaction batch sent"));
+      // There should be two "slow relay executed" transactions within the batch.
+      assert.equal(spy.getCall(-1).lastArg.mrkdwn.match(/Slow Relay executed/g).length, 2);
+
+      // Advance time such that relay has expired and check that bot correctly identifies it as expired.
+      const expirationTime = Number(relayTime.toString()) + defaultLiveness;
+      await bridgePool.methods.setCurrentTime(expirationTime).send({ from: l1Owner });
+      await l1Client.update();
+      await relayer.checkForPendingDepositsAndRelay();
+      assert.isTrue(lastSpyLogIncludes(spy, "Pending relay has expired"));
+
+      // Finally, settle. We should be able to do both of these in one tx.
+      await relayer.checkforSettleableRelaysAndSettle();
```

Comment:
Right before this line, what would happen if you sent a `deposit` that caused the Relayer to fail? Would it still be able to finalize the expired relays, despite failing to relay the deposit?

Reply:
yup! those are separate actions and do not interfere with one another. batches of txs are sent for: a) relay and b) settlement separately.

Reply:
What if the batch of `relays` throws an error?

Reply:
The part of the code where the error would be thrown has a  try catch in it. it behaves the same way that the current pr would though I belive.